### PR TITLE
remove hypershift-cli-e2e step

### PR DESCRIPTION
**Pull request by awels**
The step is causing build issues because it now requires golang 1.19 and the build is still configured to use golang 1.18. However the entire container is not needed any so we can remove the build step complete solving the build issue.

### CODE REVIEW

File: ci-operator/config/openshift/kubevirt-csi-driver/openshift-kubevirt-csi-driver-release-4.15.yaml [hunk 1]
```diff
@@ -30,36 +30,7 @@ images:
...
-      - destination_dir: .
-        source_path: /usr/bin/oc
-    hypershift-operator-4.13:
-      paths:
-      - destination_dir: .
-        source_path: /usr/bin/hypershift
-  to: hypershift-cli-e2e
 promotion:
-  excluded_images:
-  - hypershift-cli-e2e
```
Comment (davidvossel) [inline ci-operator/config/openshift/kubevirt-csi-driver/openshift-kubevirt-csi-driver-release-4.15.yaml:62]: i think you need to change `./ci-operator/step-registry/storage/conf/csi-kubevirt/storage-conf-csi-kubevirt-ref.yaml` to not reference this image anymore.

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by awels** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve
/pj-rehearse ack

**Pull request merged**